src/AzureAppConfigurationImpl.ts

```diff
@@ -74,6 +117,68 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         }
     }
 
+    /**
+     * Refresh the configuration store.
+     */
+    public async refresh(): Promise<void> {
+        if (!this.refreshEnabled) {
+            return Promise.resolve();
+        }
+
+        // if still within refresh interval/backoff, return
+        if (this.refreshTimer.isDuringBackoff()) {
+            return Promise.resolve();
+        }
+
+        // try refresh if any of watched settings is changed.
+        let needRefresh = false;
+        for (const sentinel of this.sentinels) {
+            const response = await this.client.getConfigurationSetting(sentinel, {

```

Comment:
What if any error happen while querying the sentinel key? Seems we should handle it as well.

Reply:
I agree that for RestError with 404 we should handle it, as it's a valid case when the setting is deleted.

For other unexpected errors, shall we silent fail in order not to break the app?

Reply:
We should bubble up all errors and increase the backoff time. 
In .NET, this immediately breaks the app if `RefreshAsync` is invoked. But if users are invoking `TryRefreshAsync`, app will not crash on known errors (like `RequestFailedException`, `AggregateException`, etc). App can still crash on unknown errors.
In Python, we provide an additional error callback that can be invoked if there are any known errors. If no callback is provided, or an unknown error occurs, app will crash.

Reply:
We didn't design to have an API `TryRefresh()` here right? Probably we could do similar with Python, provide an error callback that can be invoked if any known error happen

Reply:
@RichardChen820 
No, we don't provide `TryRefresh()`, as user can wrap `refresh()` with try-catch blocks on their own, and deal with different kinds of errors.

JS is a language full of callbacks. The `refresh()` returns a promise, which works with callbacks naturally.
```js
Promise.then(successCallback).catch(errorCallback);
```
Note that, we do have an dedicated `onRefresh()` API to pass success callbacks, serving for the purpose that you want to set callbacks in elsewhere, other than the place you call refresh(). Maybe if necessary, later we can have an onRefreshFailure() API for error callback...

Reply:
With that Promise pattern, I think we can safely throw the errors and have user code handle it. The only one we need to catch is 404 like you said, and make sure to increment backoff time on any error.

Reply:
404 handled in 46be338 .